### PR TITLE
add variable entropy readers to cert gen helpers [VAULT-1179]

### PR DESCRIPTION
**Pull request by HridoyRoy**
This PR goes hand in hand with hashicorp/vault-plugin-secrets-kmip#95 . It allows for private keys to be used with entropy augmentation if needed.
Moved from:https://github.com/hashicorp/vault-enterprise/pull/1664

### CODE REVIEW

**Review by sgmiller — APPROVED** [review]

File: sdk/helper/certutil/helpers.go [hunk 1]
```diff
@@ -239,7 +256,17 @@ func GeneratePrivateKey(keyType string, keyBits int, container ParsedPrivateKeyC
 
 // GenerateSerialNumber generates a serial number suitable for a certificate
 func GenerateSerialNumber() (*big.Int, error) {
-	serial, err := rand.Int(rand.Reader, (&big.Int{}).Exp(big.NewInt(2), big.NewInt(159), nil))
+	return generateSerialNumber(rand.Reader)
+}
+
+// GenerateSerialNumberWithEntropy generates a serial number suitable
+// for a certificate with custom entropy.
+func GenerateSerialNumberWithEntropy(randReader io.Reader) (*big.Int, error) {
```
Comment (sgmiller) [inline sdk/helper/certutil/helpers.go:264]: Maybe s/Entropy/RandomSource/

File: sdk/helper/certutil/helpers.go [hunk 2]
```diff
@@ -620,15 +657,25 @@ func CreateCertificate(data *CreationBundle) (*ParsedCertBundle, error) {
 
 var oidExtensionBasicConstraints = []int{2, 5, 29, 19}
 
-// Creates a CSR. This is currently only meant for use when
-// generating an intermediate certificate.
+// CreateCSR creates a CSR with the default rand.Reader to
+// generate a cert/keypair.
```
Comment (sgmiller) [inline sdk/helper/certutil/helpers.go:661]: Don't lose the bit about "This is currently only..."

**Review by sgmiller — CHANGES_REQUESTED** [review]
Just minor naming suggestions, otherwise lgtm.

File: sdk/helper/certutil/helpers.go (hunk 1 above)
Comment (HridoyRoy) [reply sdk/helper/certutil/helpers.go]: Done!

File: sdk/helper/certutil/helpers.go (hunk 2 above)
Comment (HridoyRoy) [reply sdk/helper/certutil/helpers.go]: Done!

**Review by sgmiller — APPROVED** [review]
Looks great.

**Pull request merged**